game-core/src/main/java/games/strategy/engine/data/UnitHitsChange.java

```diff
@@ -11,18 +14,42 @@
   private final IntegerMap<Unit> undoHits;
   private final Collection<Territory> territoriesToNotify;
 
+  private final IntegerMap<String> primitiveHits;
+  private final IntegerMap<String> primitiveUndoHits;
+  private final Collection<String> primitiveTerritoriesToNotify;
+
   private UnitHitsChange(
+      final IntegerMap<String> primitiveHits,
+      final IntegerMap<String> primitiveUndoHits,
+      final Collection<String> primitiveTerritoriesToNotify,
       final IntegerMap<Unit> hits,
       final IntegerMap<Unit> undoHits,
       final Collection<Territory> territoriesToNotify) {
-    this.hits = hits;
-    this.undoHits = undoHits;
-    this.territoriesToNotify = territoriesToNotify;
+    if (hits != null) {
+      this.primitiveHits = new IntegerMap<>();
+      for (final Map.Entry<Unit, Integer> hit : hits.entrySet()) {
+        this.primitiveHits.add(hit.getKey().getId().toString(), hit.getValue());
+      }
+      this.primitiveUndoHits = new IntegerMap<>();
+      for (final Map.Entry<Unit, Integer> hit : undoHits.entrySet()) {
+        this.primitiveUndoHits.add(hit.getKey().getId().toString(), hit.getValue());
+      }
+      this.primitiveTerritoriesToNotify =
+          territoriesToNotify.stream().map(Territory::getName).collect(Collectors.toList());
+    } else {
+      this.primitiveHits = primitiveHits;
+      this.primitiveUndoHits = primitiveUndoHits;
+      this.primitiveTerritoriesToNotify = primitiveTerritoriesToNotify;
+    }
+
+    this.hits = null;
+    this.undoHits = null;
```

Comment:
Codacy found an issue: [Assigning an Object to null is a code smell.  Consider refactoring.](https://app.codacy.com/gh/triplea-game/triplea/pullRequest?prid=6281351)